### PR TITLE
bans posibrains and mmis from savannah ivanov

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Savannah Ivanov cannot have mmis or posibrains in them.

## Why It's Good For The Game

Now I know that sounds BAD... but I originally intended to do this as it removes a huge trust element and bypasses recruiting people for cooperation. I simply uuuh genuinely forgot to include this, so sorry for the DRAMA.

I was considering blocking the AI too, but AIs are actually more likely to be turned against you than fellow crewmembers

## Changelog
:cl:
balance: mmis can no longer possess savannah ivanovs
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
please don't put emojii in your commit names.

**Comment by tralezab** [comment]
my bad dog

**Comment by IndieanaJones** [comment]
I'm downvoting this just because I had a blast with my human buddy in one of these mechs, eventually transporting him to medbay only to watch him die.

You also didn't add a way to remove an unconscious person from the mech, which kind of sucks.

**Comment by tralezab** [comment]
oh yeah sure i'll add that

**Comment by Rohesie** [comment]
> 
> 
> oh yeah sure i'll add that

Let us know when it's ready for review.

**Comment by Technoturnovers** [comment]
Something I don't get: what says that an MMI or posibrain has to be loyal, in any way?

**Comment by newfren** [comment]
@Technoturnovers the text given to you when you become a posibrain is very explicit.

**Comment by TheBonded** [comment]
Finally the bloodthirsty posimechs are put in their place

**Comment by Fikou** [comment]
> Something I don't get: what says that an MMI or posibrain has to be loyal, in any way?

its not in the code, but the config explicitly says:
"If placed into an exosuit, you must follow your creator's orders."

**Comment by moocowswag** [comment]
Posibrains make sense but why block a man machine interface from being used?

Antags who are put into an mmi keep their antag status and mmis dont have laws to overwrite that.

**Comment by TheBonded** [comment]
> Posibrains make sense but why block a man machine interface from being used?
> 
> Antags who are put into an mmi keep their antag status and mmis dont have laws to overwrite that.

Because posibrains are just a subtype of MMI and it would be hard to allow one without the other, probably

File: code/modules/vehicles/mecha/_mecha.dm [hunk 1]
```diff
@@ -996,32 +998,32 @@
...
-	if(!M.brain_check(user))
+	if(!brain_obj.brain_check(user))
 		return FALSE
 
-	var/mob/living/brain/B = M.brainmob
-	if(!user.transferItemToLoc(M, src))
-		to_chat(user, span_warning("\the [M] is stuck to your hand, you cannot put it in \the [src]!"))
+	var/mob/living/brain/brain_mob = brain_obj.brainmob
+	if(!user.transferItemToLoc(brain_obj, src))
+		to_chat(user, span_warning("\the [brain_obj] is stuck to your hand, you cannot put it in \the [src]!"))
```
Comment (coiax) [inline code/modules/vehicles/mecha/_mecha.dm:1013]:
```suggestion
		to_chat(user, span_warning("[brain_obj] is stuck to your hand, you cannot put it in [src]!"))
```
Byond already handles the definitive article, so these `\the`s are not required, right?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 2]
```diff
@@ -996,32 +998,32 @@
...
-	if(!M.brain_check(user))
+	if(!brain_obj.brain_check(user))
 		return FALSE
 
-	var/mob/living/brain/B = M.brainmob
-	if(!user.transferItemToLoc(M, src))
-		to_chat(user, span_warning("\the [M] is stuck to your hand, you cannot put it in \the [src]!"))
+	var/mob/living/brain/brain_mob = brain_obj.brainmob
+	if(!user.transferItemToLoc(brain_obj, src))
+		to_chat(user, span_warning("[brain_obj] is stuck to your hand, you cannot put it in [src]!"))
```
Comment (Fikou) [inline code/modules/vehicles/mecha/_mecha.dm:1013]: balooney alert?

Comment (coiax) [reply code/modules/vehicles/mecha/_mecha.dm:1013]: The other messages of this nature aren't balloon alerts, I'd prefer them to be consistent.

Comment (coiax) [reply code/modules/vehicles/mecha/_mecha.dm:1013]: But if you want to convert them all in a single pass, Fikou, I'd be onboard with that.

**Pull request merged**